pkg/deploy/controller/deployment/factory.go

```diff
@@ -122,6 +126,13 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 	}
 }
 
+// setPodAttributes updates the deployer pod attributes based on the deployment strategy configuration
+func (factory *DeploymentControllerFactory) setPodAttributes(strategy *deployapi.DeploymentStrategy, pod *kapi.Pod) {
```

Comment:
Per our other discussion, you can remove the `setPodAttributes` abstraction and just inline the `util.MergeInto` calls in the controller `makeDeployerPod` function.